src/webpub_manifest_parser/core/parsers.py

```diff
@@ -488,14 +495,24 @@ def parse(self, value):
         :return: Parsed date object
         :rtype: datetime.datetime
         """
+        value = super(DateParser, self).parse(value)
+        result = None
+
         try:
-            return datetime_parser.parse(value)
+            result = datetime_parser.isoparse(value)
         except Exception as exception:
-            raise ValueParserError(
-                value,
-                u"Value '{0}' is not a correct date".format(encode(value)),
-                exception,
-            )
+            self._raise_error(value, exception)
+
+        if result and (
+            result.hour
+            or result.minute
+            or result.second
+            or result.microsecond
+            or result.tzinfo
+        ):
+            self._raise_error(value)
```

Comment:
I'm not sure about this part. Do we want the parser to raise errors when the time part is present for `date` values?